server/src/main/java/org/elasticsearch/action/admin/cluster/stats/MappingStats.java

```diff
@@ -114,6 +117,17 @@ public static MappingStats of(Metadata metadata, Runnable ensureNotCancelled) {
         return new MappingStats(fieldTypes.values(), runtimeFieldTypes.values());
     }
 
+    private static void updateScriptParams(Object scriptSourceObject, FieldScriptStats scriptStats) {
+        if (scriptSourceObject != null) {
+            String scriptSource = scriptSourceObject.toString();
+            int chars = scriptSource.length();
+            long lines = scriptSource.lines().count();
+            int docUsages = countOccurrences(scriptSource, "doc[\\[\\.]");
+            int sourceUsages = countOccurrences(scriptSource, "params\\._source");
```

Comment:
Not a new issue but it would be nice if we could avoid recompiling the regex on every field.